virtool/analyses/checks.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime
+from typing import Optional
+
+from virtool.analyses.utils import find_nuvs_sequence_by_index
+from virtool.data.errors import (
+    ResourceConflictError,
+    ResourceNotModifiedError,
+    ResourceNotFoundError,
+)
+
+
+async def check_if_analysis_modified(
+    if_modified_since: Optional[datetime], document: dict
+):
+    if if_modified_since is not None:
+        try:
+            updated_at = document["updated_at"]
+        except KeyError:
+            updated_at = document["created_at"]
+
+        if if_modified_since == updated_at:
+            raise ResourceNotModifiedError()
+
+
+async def check_if_analysis_ready(jobs_api_flag: bool, ready: bool):
+    if jobs_api_flag and ready:
+        raise ResourceConflictError()
+    else:
+        if not ready:
+            raise ResourceConflictError()
```

Comment:
The `else` is pointless here. When your initial `if` evaluates `True`, you will immediately leave the function and never get to `else`.

Also note the existence of `elif` in Python.

```suggestion
    if jobs_api_flag and ready:
        raise ResourceConflictError()
        
    if not ready:
        raise ResourceConflictError()
```